plugins/inputs/influxdb_v2_listener/influxdb_v2_listener.go

```diff
@@ -222,12 +224,47 @@ func (h *InfluxDBV2Listener) routes() error {
 	)
 
 	h.mux.Handle("/api/v2/write", authHandler(h.handleWrite()))
+	h.mux.Handle("/api/v2/health", h.handleHealth())
 	h.mux.Handle("/api/v2/ready", h.handleReady())
+	h.mux.Handle("/health", h.handleHealth())
+	h.mux.Handle("/ready", h.handleReady())
 	h.mux.Handle("/", authHandler(h.handleDefault()))
 
 	return nil
 }
 
+func (h *InfluxDBV2Listener) handleHealth() http.HandlerFunc {
+	return func(res http.ResponseWriter, _ *http.Request) {
+		defer h.healthsServed.Incr(1)
+
+		res.Header().Set("Content-Type", "application/json")
+		body := map[string]string{
+			"name":    "telegraf",
+			"commit":  internal.Commit,
+			"message": "ready for queries and writes",
+			"status":  "pass",
+			"version": internal.Version,
+		}
+
+		pendingMetrics := h.totalUndeliveredMetrics.Load()
+		if h.MaxUndeliveredMetrics > 0 && pendingMetrics >= int64(h.MaxUndeliveredMetrics) {
+			res.WriteHeader(http.StatusServiceUnavailable)
+			body["status"] = "fail"
+			body["message"] = fmt.Sprintf("pending undelivered metrics (%d) is above limit", pendingMetrics)
+		} else {
+			res.WriteHeader(http.StatusOK)
+		}
+
+		b, err := json.Marshal(body)
+		if err != nil {
+			h.Log.Debugf("error marshalling json in handleHealth: %v", err)
+		}
+		if _, err := res.Write(b); err != nil {
+			h.Log.Debugf("error writing in handleHealth: %v", err)
+		}
```

Comment:
It doesn't make sense to write `b` if you cannot marshal the body! Shouldn't we simply return a `500` in those cases?

Reply:
Likewise. This was actually @jamesreprise ' code from before we found @Hipska's PR originally, in the first commit, but we just followed the pattern already in existence for `handleReady` below (L279-284), figuring if it had got merged for that then it would be fine here too. I must admit I didn't think about what it would mean to try and write an undefined body if you failed parsing.

If I modify this, do you want me to go and change `handleReady()` too? I must admit I'm not very used to Go so I'm not familiar with its exception handling processes. Do I need to throw or return `err` here to cause something higher up the stack to return 500 to the client, or would you prefer that was handled in this function?